File: Cpp/925_Long_Pressed_Name.cpp
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        int j = 0;
        for(int i=0; i<typed.size(); ++i){
            if(typed[i] == name[j]) {
                ++j;
            } else if(j == 0 || typed[i] != name[j-1]){
                return false;
            }
        }
        
        return name.size() == j;
    }
};

File: Kotlin/925_Long_Pressed_Name.kt
class Solution {
    fun isLongPressedName(name: String, typed: String): Boolean {
        var i = 0
        for(ch in typed){
            if(i < name.length && ch == name[i]) {
                ++i
            } else if(i == 0 || ch != name[i - 1]){
                return false
            }
        }
        
        return name.length == i
    }
}